cpp/src/io/parquet/page_enc.cu

```diff
@@ -1263,10 +1263,21 @@ class header_encoder {
   inline __device__ void set_ptr(uint8_t* ptr) { current_header_ptr = ptr; }
 };
 
+// byteswap 128 bit integer into char array in network byte order
+static __device__ void swap128(__int128_t v, void* dst)
```

Comment:
Why is `dst` a `void*`? The docstring says "into char array", so shouldn't it be `char*`? Better yet, can we use `std::byte` here everywhere you have `unsigned char` and `void`?

Reply:
Also, would `byte_reverse128` be a better name for this function? `swap` implies exactly two items, but there are 16 bytes in an `int128`.

Reply:
> Why is `dst` a `void*`? The docstring says "into char array", so shouldn't it be `char*`? Better yet, can we use `std::byte` here everywhere you have `unsigned char` and `void`?

The scratch buffer is allocated elsewhere and passed around as a void * since it can be used for float scratch as well.  Changing the signature here just shifts where the cast must be performed.  Will change the comment since it wasn't changed when dst changed from char to void.

> Also, would `byte_reverse128` be a better name for this function? `swap` implies exactly two items, but there are 16 bytes in an `int128`.

Changing endianness has always been called byte swapping, even after words exceeded 16 bits ;-).  Regardless of the implementation, bytes 0 and 15 are swapped, and 1 and 14 etc.

Reply:
I looked earlier into using `std::byte*` for the scratch buffers in cuIO and it seemed like we need to make wide changes to make good use of it. I'm fine with the use of `void*` here as it's consistent with the existing code and we can update to `std::byte` as part of a future refactoring.